File: Data Structures & Algorithms/design-hashset/submission-0.cpp
class MyHashSet {
public:
    unordered_map<int,int>mp;
    MyHashSet() {
        
    }
    
    void add(int key) {
        if(!contains(key))mp[key]++;
    }
    
    void remove(int key) {
        if(contains(key))mp.erase(key);
    }
    
    bool contains(int key) {
        if(mp.find(key)!=mp.end())return true;
        else{
            return false;
        }
    }
};

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet* obj = new MyHashSet();
 * obj->add(key);
 * obj->remove(key);
 * bool param_3 = obj->contains(key);
 */

File: Data Structures & Algorithms/maximum-frequency-stack/submission-0.py
from collections import defaultdict


class FreqStack:
    def __init__(self):
        self.freqMap = defaultdict(int)
        self.grpMap = defaultdict(list)
        self.maxFreq = 0

    def push(self, val: int) -> None:
        self.freqMap[val] += 1
        self.maxFreq = max(self.maxFreq, self.freqMap[val])
        self.grpMap[self.freqMap[val]].append(val)

    def pop(self) -> int:
        num = self.grpMap[self.maxFreq].pop()
        if not self.grpMap[self.maxFreq]:
            self.maxFreq -= 1
        self.freqMap[num] -= 1
        return num


# Your FreqStack object will be instantiated and called as such:
# obj = FreqStack()
# obj.push(val)
# param_2 = obj.pop()
